src/infra/cache/index.ts

```diff
@@ -0,0 +1,93 @@
+import * as cacheManager from 'cache-manager';
+import redisStore = require('cache-manager-redis-store');
+import primitiveKey = require('memoizee/normalizers/primitive');
+import { REDIS_HOST, REDIS_PORT } from '../../lib/config';
+import { promisify } from 'util';
+import { version } from '../../../package.json';
+
+/**
+ * A multi layer cache compatible with a subset of memoizee options
+ */
+export const multiCacheMemoizee = <T extends (...args: any[]) => any>(
+	fn: T,
+	opts: {
+		cacheKey?: string;
+		promise: true;
+		primitive: true;
+		preFetch?: true | number;
+		maxAge: number;
+		max?: number;
+	},
+) => {
+	const {
+		cacheKey = fn.name,
```

Comment:
How about enforcing a `featureName.fnName` pattern?

Reply:
I don't think it's possible to enforce as there's no way of knowing the feature name, beyond that this is "infra" code so shouldn't really have any specific knowledge and ideally will get moved out of open-balena-api altogether as it should be generically useful